test/fixtures/configSchemaExtensionsError/test-plugin-with-errors.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+class TestPluginWithErrors {
+  constructor(serverless) {
+    serverless.configSchemaHandler.defineProvider('someProvider', {
+      function: {
+        properties: {
+          handler: { type: 'string' },
+          events: {
```

Comment:
Function events should be passed through `functionEvents` option.

Passing events that way seems error-prone (I see it as an issue that `defineProvider` didn't crash - I believe we should improve those methods so they do not allow overwrites - just opened an issue to address that: https://github.com/serverless/serverless/issues/8481)